lightning/src/chain/keysinterface.rs

```diff
@@ -738,7 +738,7 @@ impl Readable for InMemorySigner {
 /// Cooperative closes may use seed/2'
 /// The two close keys may be needed to claim on-chain funds!
 pub struct KeysManager {
-	secp_ctx: Secp256k1<secp256k1::SignOnly>,
+	secp_ctx: Secp256k1<secp256k1::All>,
```

Comment:
Was this change deliberate? It doesn't seem to go with the commit message.

Reply:
Yes.  I'll add to the commit message.

It just doesn't seem worth it to deal with both kinds of contexts separately, given that the caller might as well have an `All` handy to be able to supply both for different calls.

I think it's also not possible to pass in an `All` when `Signing` is expected without generics, which we are trying to get away from here.